pkgs/by-name/gh/ghostty/package.nix

```
@@ -0,0 +1,138 @@
+{
+  lib,
+  stdenv,
+  bzip2,
+  callPackage,
+  expat,
+  fetchFromGitHub,
+  fontconfig,
+  freetype,
+  git,
+  glib,
+  gsettings-desktop-schemas,
+  gtk4,
+  harfbuzz,
+  libadwaita,
+  libGL,
+  libpng,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  ncurses,
+  nix-update-script,
+  oniguruma,
+  pandoc,
+  pkg-config,
+  wrapGAppsHook4,
+  zig_0_13,
+  zlib,
+}:
+let
+  # Ghostty needs to be built with --release=fast, --release=debug and
+  # --release=safe enable too many runtime safety checks.
+  zig_hook = zig_0_13.hook.overrideAttrs {
+    zig_default_flags = "-Dcpu=baseline -Doptimize=ReleaseFast --color off";
+  };
+in
+stdenv.mkDerivation (final: {
+  pname = "ghostty";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ghostty-org";
+    repo = "ghostty";
+    rev = "v${final.version}";
+    hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
+  };
+
+  nativeBuildInputs = [
+    git
+    ncurses
+    pandoc
+    pkg-config
+    wrapGAppsHook4
+    zig_hook
+  ];
+
+  buildInputs = [
+    bzip2
+    expat
+    fontconfig
+    freetype
+    glib
+    gsettings-desktop-schemas
+    gtk4
+    harfbuzz
+    libadwaita
+    libGL
+    libpng
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    oniguruma
+    zlib
+  ];
+
+  dontPatch = true;
+  dontConfigure = true;
+  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
+  doCheck = false;
+
+  deps = callPackage ./deps.nix { name = "${final.pname}-cache-${final.version}"; };
+
+  zigBuildFlags = [
+    "--system"
+    "${final.deps}"
+    "-Dversion-string=${final.version}"
+  ];
+
+  outputs = [
+    "out"
+    "terminfo"
+    "shell_integration"
+    "vim"
+  ];
+
+  checkPhase = ''
+    zig build --verbose -Dcpu=baseline -Doptimize=Debug ${lib.strings.concatStringsSep " " final.zigBuildFlags} test
+  '';
+
+  postInstall = ''
+    mkdir -p "$terminfo/share"
+    mv "$out/share/terminfo" "$terminfo/share/terminfo"
+    ln -sf "$terminfo/share/terminfo" "$out/share/terminfo"
+
+    mkdir -p "$shell_integration"
+    mv "$out/share/ghostty/shell-integration" "$shell_integration/shell-integration"
+    ln -sf "$shell_integration/shell-integration" "$out/share/ghostty/shell-integration"
+
+    mv "$out/share/vim/vimfiles" "$vim"
+    ln -sf "$vim" "$out/share/vim/vimfiles"
+  '';
+
+  postFixup = ''
+    patchelf --add-rpath "${lib.makeLibraryPath [ libX11 ]}" "$out/bin/.ghostty-wrapped"
```

Should be preFixup so that we don't have to refer to .ghostty-wrapped, but also maybe you can set `dontWrapGApps = true;` and do

```suggestion
    wrapProgram $out/bin/ghostty \
      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libX11 ]}" \
      "gappsWrapperArgs[@]"
```

---

```nix
    preFixup = ''
      gappsWrapperArgs+=(--prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libX11 ]}")
    '';
```

Looks like the recommendation: https://github.com/NixOS/nixpkgs/blob/edc08c763858607e2e627863908ac5360981f071/doc/languages-frameworks/gnome.section.md?plain=1#L129-L143

---

That also works, yup